cclient/http/http.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "http.h"
+#include <string.h>
+#include "cclient/service.h"
+#include "http_parser.h"
+#include "utils/macros.h"
+#include "utils/sockets.h"
+
+#define RECEIVE_BUFFER_SIZE 1024
```

Comment:
Ok, but just guessing here. I've moved the `#define` to `utils/sockets.h`. A better solution would be to pickup the current socket receive size from the system at run time and allocate the buffer dynamically. The problem: MSVC doesn't support VLA so we need to use `malloc` - but maybe it would be acceptable to `#ifdef` the code just for Windows?

Reply:
Agree. We've started using malloc in a lot of other places anyway, it's sane to do that here as well I guess. We'll pay the overhead of that later if need be.

Reply:
Could you file an issue for getting a better receive buffer size dynamically?